### PR TITLE
Change npm package scope

### LINKED ISSUE #2426: Change npm packages to @hashgraph scope
### Problem

While no one uses our packages as dependencies since they're applications and we don't publish/pull our packages to/from private registries, we should still ensure users aren't confused by public packages with the same name that are not related.

### Solution

- Change `hedera-mirror-rest` to `@hashgraph/mirror-rest`
- Change `hedera-mirror-monitor` to `@hashgraph/mirror-monitor`
- Change any docs that reference old package name
- Rename `hashgraph-mirror-rest-*tgz` produced from `npm pack` back to `hedera-mirror-rest-*.tgz` so we don't break ops automation

### Alternatives

Claim public npm packages

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

- change package scope to @hashgraph for `hedera-mirror-rest`, `check-state-proof`, and `monitor`
- change github workflow to rename hashgraph-mirror-rest-{version}.tgz to hedera-mirror-rest-v{version}.tgz
 
**Related issue(s)**:

Fixes #2426 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/check-state-proof/package-lock.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "hedera-mirror-state-proof-demo",
+  "name": "@hashgraph/mirror-state-proof-demo",
```
Comment (steven-sheehy) [inline hedera-mirror-rest/check-state-proof/package-lock.json:2]: Should we rename package to match folder? `@hashgraph/check-state-proof`?

File: hedera-mirror-rest/check-state-proof/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "hedera-mirror-state-proof-demo",
+  "name": "@hashgraph/mirror-state-proof-demo",
   "version": "1.0.0",
   "description": "An example implementation of a state proof flow fora transaction utilizing the mirror node.",
```
Comment (steven-sheehy) [inline hedera-mirror-rest/check-state-proof/package.json:4]:
```suggestion
  "description": "An example implementation of a state proof flow for a transaction utilizing the mirror node.",
```

**Review by steven-sheehy — DISMISSED** [review]
LGTM

File: hedera-mirror-rest/check-state-proof/package-lock.json (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-rest/check-state-proof/package-lock.json]: made the change

**Review by steven-sheehy — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**